extension/chrome/settings/modules/security.ts

```diff
@@ -7,122 +7,135 @@ import { Store } from '../../../js/common/platform/store.js';
 import { Ui } from '../../../js/common/browser.js';
 import { Settings } from '../../../js/common/settings.js';
 import { Api } from '../../../js/common/api/api.js';
-import { Backend } from '../../../js/common/api/backend.js';
+import { Backend, FcUuidAuth } from '../../../js/common/api/backend.js';
 import { Assert } from '../../../js/common/assert.js';
 import { initPassphraseToggle } from '../../../js/common/ui/passphrase_ui.js';
 import { Xss } from '../../../js/common/platform/xss.js';
 import { Url } from '../../../js/common/core/common.js';
+import { View } from '../../../js/common/view.js';
+import { KeyInfo } from '../../../js/common/core/pgp.js';
+
+View.run(class SecurityView extends View {
+  private readonly acctEmail: string;
+  private readonly parentTabId: string;
+  private readonly embedded: boolean;
+  private primaryKi: KeyInfo | undefined;
+  private authInfo: FcUuidAuth | undefined;
+
+  constructor() {
+    super();
+    const uncheckedUrlParams = Url.parse(['acctEmail', 'embedded', 'parentTabId']);
+    this.acctEmail = Assert.urlParamRequire.string(uncheckedUrlParams, 'acctEmail');
+    this.parentTabId = Assert.urlParamRequire.string(uncheckedUrlParams, 'parentTabId');
+    this.embedded = uncheckedUrlParams.embedded === true;
+  }
 
-Catch.try(async () => {
-
-  const uncheckedUrlParams = Url.parse(['acctEmail', 'embedded', 'parentTabId']);
-  const acctEmail = Assert.urlParamRequire.string(uncheckedUrlParams, 'acctEmail');
-  const parentTabId = Assert.urlParamRequire.string(uncheckedUrlParams, 'parentTabId');
-  const embedded = uncheckedUrlParams.embedded === true;
-
-  await initPassphraseToggle(['passphrase_entry']);
+  async render() {
+    await initPassphraseToggle(['passphrase_entry']);
+    [this.primaryKi] = await Store.keysGet(this.acctEmail, ['primary']);
+    Assert.abortAndRenderErrorIfKeyinfoEmpty(this.primaryKi, false);
+    if (!this.primaryKi) {
+      return; // added do_throw=false above + manually exiting here because security.htm can indeed be commonly rendered on setup page before setting acct up
```

Comment:
Actually this is no longer true, we don't render security.htm before setting up acct anymore. You can remove the `false` above, and remove this if/return

Reply:
Okay, got it,
Done